File: OkCompu/backend/backend-okcompu/src/main/java/com/okcompu/ecommerce/backendokcompu/service/VentaService.java
package com.okcompu.ecommerce.backendokcompu.service;

import com.okcompu.ecommerce.backendokcompu.dto.VentaDTO;
import com.okcompu.ecommerce.backendokcompu.model.Venta;

import java.time.LocalDateTime;
import java.util.List;

public interface VentaService extends CRUD<Venta, Long> {

    Venta registrarVenta(VentaDTO ventaDTO);
    Venta anularVenta(Long id);
    List<Venta> buscarPorFechas(LocalDateTime fechaInicio, LocalDateTime fechaFin);
    List<Venta> buscarPorCliente(Long idCliente);
}

File: OkCompu/frontend/frontend-okcompu/src/app/services/venta.service.ts
import { Injectable } from '@angular/core';
import { Venta } from '../model/venta';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment.development';
import { GenericService } from './generic.service';
import { VentaFiltro } from '../model/venta-filtro';

@Injectable({
  providedIn: 'root'
})
export class VentaService extends GenericService<Venta> {

  constructor(protected override http: HttpClient) {
      super(http, `${environment.HOST}/ventas`)
    }

    buscarPorFechas(filtro: VentaFiltro) {  
      return this.http.get<Venta[]>(`${this.url}/fechas`, {  
        params: {  
          fechaInicio: filtro.fechaInicio,  
          fechaFin: filtro.fechaFin  
        }  
      });  
    }  
  
    buscarPorCliente(idCliente: number) {  
      return this.http.get<Venta[]>(`${this.url}/cliente/${idCliente}`);  
    } 

}
